gitpod-io/gitpod
# [Server] Extract IDE config resolution logic to ide-service

## Description

This PR allows to migrate resolution of IDE configuration for each workspace from server to ide-service. It is done by using configcat flag. By default is disabled, but the request to ide-service is done anyway that we could confirm reliability and functional correctness, before we migrate.

IDE service part is already deployed and tested: [[1](https://github.com/gitpod-io/gitpod/pull/15118)]

Co-Authored between: @akosyakov @iQQBot @felladrin @andreafalzetti 

## Related Issue(s)

Related to #9189

## How to test
Check if the IDE opened is the right version
In the server check if the asser log never appears for the following cases:

- [x] Start workspaces for VS Code Stable/Latest -> @iQQBot 
- [x] Start workspaces for IntelliJ Stable/Latest -> @akosyakov 
- [x] Start workspaces for IntelliJ from Gateway Stable/Latest -> @felladrin 
- [x] Test prebuilds (https://github.com/andreafalzetti/gitpod-prebuilds/) -> @andreafalzetti

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
